backend/app/api/endpoints/knowledge.py

```
@@ -767,3 +772,124 @@ def get_qa_history(
+        result = SummaryService.trigger_document_summary(
+            db=db,
+            document_id=document_id,
+            user_id=current_user.id,
+            force=data.force,
+        )
+        return result
+    except ValueError as e:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail=str(e),
+        )
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)